GpsApp/Top/Main.cpp

```diff
@@ -18,16 +18,38 @@ static void sighandler(int signum) {
     terminate = 1;
 }
 
+void run1cycle(void) {
+    // call interrupt to emulate a clock
+    blockDrv.callIsr();
+    Os::Task::delay(1000); //1Hz
+}
+
+void runcycles(NATIVE_INT_TYPE cycles) {
+    if (cycles == -1) {
+        while (true) {
+            run1cycle();
+        }
+    }
+
+    for (NATIVE_INT_TYPE cycle = 0; cycle < cycles; cycle++) {
+        run1cycle();
+    }
+}
+
 int main(int argc, char* argv[]) {
-    I32 option = 0;
-    U32 port_number = 50000;
-    char* hostname = NULL;
-    char* device = (char*)"/dev/ttyUSB0";
+    U32 port_number = 0; // Invalid port number forced
+    I32 option;
+    char *hostname;
+    char *device;
+    option = 0;
+    hostname = NULL;
+    device = (char*)"/dev/ttyUSB0";
+    bool dump = false;
```

Comment:
One last tiny thing I noticed doing a quick local test, I got a compiler warning that `dump` was unused... do you mind removing the variable?